sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -100,7 +105,9 @@ object SQLExecution {
       // all accumulator metrics will be 0. It will confuse people if we show them in Web UI.
       //
       // A real case is the `DataFrame.count` method.
-      throw new IllegalArgumentException(s"$EXECUTION_ID_KEY is already set")
+      throw new IllegalArgumentException(s"$EXECUTION_ID_KEY is already set, please wrap your " +
```

Comment:
`SQLExecution` is kind of a developer API, people who develop data source may need to call `ignoreNestedExecutionId` inside their data source implementation, as reading/writing data source will be run inside a command and they may hit the nested execution problem. What do you think?